backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -309,16 +264,41 @@ private Project insertProject(Project project, String workspaceName, UserInfo us
     }
   }
 
+  private void removeDeletedProjectByNameFromUAC(String projectName, Session session) {
+    Query deletedEntitiesQuery = session.createQuery(GET_DELETED_PROJECTS_IDS_BY_NAME_HQL);
+    deletedEntitiesQuery.setParameter("projectName", projectName);
+    List<String> deletedEntityIds = deletedEntitiesQuery.list();
+    if (!deletedEntityIds.isEmpty()) {
+      try {
+        CommonUtils.registeredBackgroundUtilsCount();
+        roleService.deleteEntityResourcesWithServiceUser(
+            deletedEntityIds, ModelDBServiceResourceTypes.PROJECT);
+      } finally {
+        CommonUtils.unregisteredBackgroundUtilsCount();
+      }
+    }
+  }
+
   @Override
   public Project updateProjectName(UserInfo userInfo, String projectId, String projectName)
```

Comment:
This method can lead to inconsistency where MDB thinks the name is one thing and UAC thinks it's a different name.

Reply:
This will remove in different PR.

Reply:
PR: https://github.com/VertaAI/modeldb/pull/1945